### PR TITLE
constructors-in-kotlin[duplicate]

### LINKED ISSUE #4978: [KOTLIN ] Constructors and Init blocks in Kotlin
#### Proposed title of article
 Constructors and Init blocks in Kotlin

Use keyword research to improve your article's SEO.

#### Proposed article introduction
Effective Kotlin reguires the use of classes we are going to talk about how to Instatiate objects using Constructors and the init Block
 The Constructor will initialize new objects or instances of the class and  we are going to cover the Best practices in object oriented programming

#### Key takeaways 
1. Understanding init block in kotlin
2. How to instantiate objects using constructors 
3. Best practices for object oriented programming(OOP)

#### Article quality
I am going to produce an article that is well understandable by the users making them understand what does the Init block do in Kotlin and when to use the it

#### References
N/A

### PR DESCRIPTION
**Pull request by kamothosimon**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4978

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

File: content/articles/constructors-in-kotlin/index.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+```
+The output of the above code is:
+
+```
+Name = Alex 
+Id = 200
+Email = Alex@gmail
+```
+
+### Init blocks
```
Comment (john-amiscaray) [inline content/articles/constructors-in-kotlin/index.md:133]: You never properly explained when to use the init block despite saying you would in the intro. It sounds vague as to why you would use this over a secondary constructor.

File: content/articles/constructors-in-kotlin/index.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+- The Java Development Kit (JDK) installed on your computer
+- A basic understanding of programming languages
+- The IntelliJ code editor
+
+### What is a constructor?
+A constructor is a block of code that is similar to a procedure or a method, usually declared alongside the class. They are mostly used when you need to initialize variables while creating an object.
+
+### How to use constructors to create an object
+As a function, the primary constructor must be defined with the same name as the class, followed by parentheses. However, the secondary constructor must be expressed with the `constructor` keyword.
```
Comment (john-amiscaray) [inline content/articles/constructors-in-kotlin/index.md:23]: >As a function, the primary constructor must be defined with the same name as the class, followed by parentheses.

This sounds misleading. It sounds as though you are saying that it is a function within the class with the same name of said class. Instead, as you pointed out, it is located in the class declaration.

File: content/articles/constructors-in-kotlin/index.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+#### Primary constructors
+The primary constructor is a section of the class header that comes after the class name. The constructor keyword is used after the class name. The example below exemplifies the definition:
+
+```kotlin
+class student Constructor(name:string,age:Int,Email:string)
+ { 
+     ///code to be processed
+}
+```
+The constructor keyword can be ignored if the primary constructor has no annotations or accessibility modifiers:
```
Comment (john-amiscaray) [inline content/articles/constructors-in-kotlin/index.md:40]: > The constructor keyword can be ignored if the primary constructor has no annotations or accessibility modifiers:

It seems like you are missing a code snippet below this sentence.

File: content/articles/constructors-in-kotlin/index.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+```
+
+The output of the above code will look like this:
+
+```
+Name = Alex 
+Id = 200
+```
+
+Kotlin also offers room for using a primary constructor with a secondary constructor in the same class. You need to use the `this` keyword for authorization:
```
Comment (john-amiscaray) [inline content/articles/constructors-in-kotlin/index.md:84]: > You need to use the `this` keyword for authorization

Not very descriptive of what is happening here. The `this` keyword is to call the primary constructor. You should mention this.

File: content/articles/constructors-in-kotlin/index.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+- Minimize visibility of properties, i.e., private, public, protected. It allows you to control where your class members can be accessed from.
+- Combine property declaration. This is achieved by taking a property and setting it to a constructor parameter.
+- Improve variable naming. This is achieved by using variable naming rules; your variable names should: 
+  - Be based on the variable's purpose so it's clear what the variable is for.
+  - Have no spaces between words. Ex. myClass.
+  - Be greater than a single character.
+  - Not use special prefixes or suffixes (e.g. name_, mName, s_name, and kName), except in the case of backing properties.
+
+### Conclusion
+In this article, we have covered various ways of creating constructors using Kotlin as our language, and when to use the init block. We also learned that the init block is called just after the primary constructor, and the secondary constructor is called using the `this` keyword. Finally, we went through best practices in object-oriented programming.
```
Comment (john-amiscaray) [inline content/articles/constructors-in-kotlin/index.md:180]: You never explicitly said that the init block is called just after the primary constructor. While you did use a code snippet above to demonstrate it, you shouldn't assume that the reader will pick it up just by looking at the output. Please add a short explanation detailing that after showing the output (the output I'm talking about is on line 161).

Comment (john-amiscaray) [reply content/articles/constructors-in-kotlin/index.md:180]: I believe there is a mistake here: you said:

> The secondary constructor is called using the `this` keyword.

You never mentioned this above and I don't think that is correct. Did you mean the primary constructor is called using the `this` keyword?

**Review by john-amiscaray — CHANGES_REQUESTED** [review]
Please go through these new suggestions and try to implement them as best as you can and then I can give this article further reviews. Once again, I will be closing this pull request while you do so. Please open a new pull request afterward

**Comment by john-amiscaray** [comment]
Closing the pull request while you make the necessary changes.

**Comment by kamothosimon** [comment]
@john-amiscaray hello `ve the changes you can go ahead to check it.

**Pull request closed without merging**